From mtosatti@redhat.com  Thu Sep  3 14:34:01 2009
From: Avi Kivity <avi@qumranet.com>
Date: Thu,  6 Aug 2009 14:39:47 -0300
Subject: KVM: MMU: Flush tlbs after clearing write permission when accessing dirty log
To: stable@kernel.org
Cc: avi@redhat.com
Message-ID: <1249580407-21883-8-git-send-email-mtosatti@redhat.com>


From: Avi Kivity <avi@qumranet.com>

(cherry picked from commit 171d595d3b3254b9a952af8d1f6965d2e85dcbaa)

Otherwise, the cpu may allow writes to the tracked pages, and we lose
some display bits or fail to migrate correctly.

Signed-off-by: Avi Kivity <avi@qumranet.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 arch/x86/kvm/mmu.c |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -2069,6 +2069,7 @@ void kvm_mmu_slot_remove_write_access(st
 			if (pt[i] & PT_WRITABLE_MASK)
 				pt[i] &= ~PT_WRITABLE_MASK;
 	}
+	kvm_flush_remote_tlbs(kvm);
 	spin_unlock(&kvm->mmu_lock);
 }
 
